#!/bin/sh
set -e

patch="/usr/bin/patch"
package="opflexagent"

if ! [ -x "$patch" ] ; then
    echo "ERROR: Executable '$patch' not installed." 1>&2
    echo "ERROR: Please install '$patch' using 'yum install patch' or equivalent" 1>&2
    exit 1
fi

rootdir=`python -c "import $package; print $package.__path__[0]" 2>/dev/null || :`
if [ -z "$rootdir" ] ; then
    echo "ERROR: Package '$package' not installed on this host"
    exit 1
fi

echo "INFO: Patching $package at $rootdir"
cd $rootdir
$patch -p 2 <<'EOF'
From 06bd35ed2564a3ca934594eceddb6e05031d1f83 Tue May 09 04:47:12 2017
Subject: Evaluate the endpoint request timeout in seconds instead of milliseconds
---
 opflexagent/utils/port_managers/async_port_manager.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/opflexagent/utils/port_managers/async_port_manager.py b/opflexagent/utils/port_managers/async_port_manager.py
index c39cf85..f35b777 100644
--- a/opflexagent/utils/port_managers/async_port_manager.py
+++ b/opflexagent/utils/port_managers/async_port_manager.py
@@ -76,7 +76,7 @@ class AsyncPortManager(base.PortManagerBase, rpc.OpenstackRpcMixin):
         self._setup_rpc()
         self.pending_requests = RequestMap()
         self.response_by_device_id = {}
-        self.request_timeout = config['endpoint_request_timeout']
+        self.request_timeout = config['endpoint_request_timeout'] * 1000
         self.host = host
         return self
 
@@ -167,5 +167,5 @@ class AsyncPortManager(base.PortManagerBase, rpc.OpenstackRpcMixin):
             LOG.debug("Got response for port %(port_id)s in "
                       "%(secs)s seconds",
                       {'port_id': detail.get('device'),
-                       'secs': ((time.time() * 1000) -
-                                float(detail.get('timestamp', 0)))})
+                       'secs': (((time.time() * 1000) -
+                                 float(detail.get('timestamp', 0))) / 1000)})
-- 
EOF

echo "INFO: Restarting neutron-opflex-agent ..."
systemctl restart neutron-opflex-agent.service

echo "INFO: Patch applied successfully"
